Source/DafnyCore/ProofDependencyWarnings.cs

```diff
@@ -42,15 +45,93 @@ public static void WarnAboutSuspiciousDependenciesForImplementation(DafnyOptions
     if (results.Any(r => r.Outcome != SolverOutcome.Valid)) {
       return;
     }
+    var distilled = results.Select(r => (r.CoveredElements, DafnyConsolePrinter.DistillVCResult(r))).ToList();
+    var asserts = distilled.Select(tp => (tp.CoveredElements, new HashSet<(Boogie.IToken Tok, string Description, string Id)>(tp.Item2.Asserts))).ToList();
```

Comment:
Could you introduce a record `AssertCmdPartialCopy` for `(Boogie.IToken Tok, string Description, string Id)` ?

And rename `VCResultLogEntry` to `VerificationRunResultPartialCopy` ?